mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala

```diff
@@ -55,25 +55,40 @@ class MulticlassMetrics @Since("1.1.0") (predictionAndLabels: RDD[_ <: Product])
         throw new IllegalArgumentException(s"Expected Row of tuples, got $other")
     })
 
-  private lazy val labelCountByClass: Map[Double, Double] =
-    predLabelsWeight.map {
-      case (_: Double, label: Double, weight: Double) =>
+  private lazy val labelCountByClass: Map[Double, Double] = {
+    confusions.toSeq.map {
```

Comment:
Because basically any method in this class requires computing `confusions`, I wonder if it's better to just compute `confusions` and the derived values here too immediately in the constructor and not retain `predLabelsWeight`. You can do that in one pass over `confusions` with some mutable Maps too, if perf is a concern